### PR TITLE
UX-440 Add TimelineTabs component

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Adds a new component `TimelineTabs`
- Updates eslint, was getting weird errors during commit staging

### How To Test or Verify
- Run storybook
- Test the component here: http://localhost:9001/?path=/story/timelinetabs--test
- Verify keyboard events
- Verify `aria` attributes and other html attributes are correct. See https://dequeuniversity.com/library/aria/tabpanel for a reference

### PR Checklist

- [x] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,48 +4,24 @@
...
+  "settings": {
+    "react": {
+      "version": "detect"
+    }
+  },
   "rules": {
     // doesn't work with jest-in-case
     "jest/no-standalone-expect": "off",
-
-    "lodash/chaining": ["error", "never"],
```
Comment (jonambas) [inline .eslintrc:11]: Removed these rules since we don't use lodash

File: .eslintrc [hunk 2]
```diff
@@ -4,48 +4,24 @@
...
+    ],
+    "react/prop-types": 1
   },
-  "extends": [
-    "react-app",
-    "plugin:lodash/canonical",
-    "plugin:jest/recommended",
-    "plugin:prettier/recommended"
-  ],
+  "extends": ["plugin:react/recommended", "plugin:jest/recommended", "plugin:prettier/recommended"],
```
Comment (jonambas) [inline .eslintrc:24]: Adds this plugin `react/recommended`

File: packages/matchbox/src/components/TimelineTabs/Tab.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      display: none;
+    }
+  }
+`;
+
+const FocusSvg = styled.div`
+  width: 24px;
+  height: 24px;
+
+  ${Container}:focus-visible & {
```
Comment (jonambas) [inline packages/matchbox/src/components/TimelineTabs/Tab.js:58]: Need to add a fallback for this for Safari

**Comment by jonambas** [comment]
Need to rebase and rewrite stories with libby

File: packages/matchbox/src/components/TimelineTabs/TimelineTabs.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  const systemProps = pick(rest, margin.propNames);
+
+  const tabs = React.Children.toArray(children);
+  const tabRefs = React.useRef({ current: [] });
+  const focusRef = React.useRef();
+
+  const [selectedIndex, setSelectedIndex] = React.useState(initialIndex || 0);
+
+  // Preps array of tab item refs
+  React.useEffect(() => {
```
Comment (logansparlin) [inline packages/matchbox/src/components/TimelineTabs/TimelineTabs.js:25]: Could make this a hook instead?

**Review by logansparlin — APPROVED** [review]
Looks really good, just one small comment

**Pull request merged**